### DISCUSSION TITLE
Laravel Livewire checkbox values not returning from database

### DISCUSSION THREAD

**Discussion by ItsJoshDaly** [livewire/livewire#4150] [1. Help]
We're trying to fetch values from the database so that livewire can determine dynamically whether or not to render a checkbox as being selected.

``` <?php

namespace App\Http\Livewire;

use Livewire\Component;

class CheckboxInput extends Component
{
    public $checked;
    public $element;
    public $checkbox = [];
    public $component_values = [];

    public function saveButton() {
        foreach($this->element->surveyTypeComponent->options as $key => $value) {
            foreach ($value as $item) {
                $this->component_values[] = $item;
            }
        }

        foreach($this->component_values as $option) {
            if(in_array($option, $this->checkbox)) {
                $this->checked[$option] = true;
            } else {
                $this->checked[$option] = false;
            }
        }

        $temp_array = [];
        foreach ($this->element->options as $element_key => $element_option) {
            $temp_array[$element_key]['title'] = $element_option['title'];
            if (in_array($element_option['title'], $this->checked)) {
                $temp_array[$element_key]['selected'] = $this->checked[$element_option['title']];
            } else {
                $temp_array[$element_key]['selected'] = false;
            }
        }

        $this->element->options = $temp_array;
        $this->element->save();
    }

    public function render()
    {
        return view('livewire.checkbox-input');
    }
}```

``` <form>
    <div class="bg-white">
        @foreach($element->options as $checkbox_input)
            <label class="border-2 p-3 mr-1" for="checkbox_input">
                <input wire:model="checkbox" type="checkbox" value="{{ $checkbox_input['title'] }}">
                <span>{{ $checkbox_input['title'] }}</span>
            </label>
        @endforeach
        <button class="btn btn-approve" wire:click="saveButton">Save Changes</button>
    </div>
</form>```

We previously attempted to do this by using an embedded if statement inside the input tag, however, the wire:model was stopping it from working. This did work when we removed wire:model="checkbox", but this prevented us from saving.

Any help is muchly appreciated.

**Comment by joshhanley** [comment]
@ItsJoshDaly you should load your database values and pre-populate `public $checkbox = []` with the values in your `mount` method. That way it will display what is currently checked or not.

Hope this helps!

**Reply by ItsJoshDaly** [reply]
Hi there, apologies for the slow reply.

I had attempted to set ```public $checkbox = []``` using the following code.
```
        $selecteds = array_column($element->options, 'selected','selected');
        $this->checkbox = $selecteds;
``` 
So, the anticipated result would have been that we'd have ended up with an array of 'selected' containing the values of 'true' or 'false' populating checkbox in mount(). However, while we can see that the data does return a list of 'true' and 'false' values, checkbox seemingly doesn't update and doesn't show as ticked. Is this because I'm targeting those values incorrectly? Thanks.

**Reply by ItsJoshDaly** [reply]
bumping because this is urgent

**Reply by ItsJoshDaly** [reply]
bumping again.

**Reply by joshhanley** [reply]
@ItsJoshDaly hmm I'm not exactly clear what is going on. Can you share a dump of what `$element->options` looks like and also what `$selecteds` looks like once it has been created?

**Reply by ItsJoshDaly** [reply]
Well, a dump on $selecteds simply returns the following.

```array:1 [▼
  1 => true
]```

While a dump on $element->options gets me...

```array:3 [▼
  0 => array:2 [▼
    "title" => "I1: Regulation"
    "selected" => true
  ]
  1 => array:2 [▼
    "title" => "I2: Guarantee"
    "selected" => true
  ]
  2 => array:2 [▼
    "title" => "F1: Further Investigation"
    "selected" => true
  ]
]```

Now, that data inside $element->options is correct, and it will save correctly in the back-end, but if you are to save the data after making alterations, this does not reflect on the front-end.

**Reply by joshhanley** [reply]
@ItsJoshDaly thanks!

The issue is your mapping for the checkbox array needs to contain an identifier, just having a list of true/false isn't enough to remap to your specific options after.

But, as you already have a boolean in the options array, you could just bind to that directly, instead of using the checkbox property.

Just create a public property called options, and assign the options array to it on mount, then re-assign it back to the element on save.

You can then bind to the options using this

```blade
<form>
    <div class="bg-white">
        @foreach($options as $key => $option)
            <label class="border-2 p-3 mr-1" for="checkbox_input">
                <input wire:model="options.{{ $key }}.selected" type="checkbox">
                <span>{{ $option['title'] }}</span>
            </label>
        @endforeach
        <button class="btn btn-approve" wire:click="saveButton">Save Changes</button>
    </div>
</form>
```

Also for future reference, when you paste code into github, you need to surround it in triple back ticks and then specify the file type like php or blade. But the backticks (and file type) need to be on new lines above and below your code, not on the same line.

For example
\```php
\\ PHP code here
\```

\```blade
\\ HTML and blade code here
\```

Hope this helps!

**Reply by ItsJoshDaly** [reply]
That actually did help, thank you very much :)

**Reply by joshhanley** [reply]
No worries! 🙂
